pkg/cluster/internal/providers/docker/provision.go

```diff
@@ -320,6 +328,79 @@ func getSubnets(networkName string) ([]string, error) {
 	return strings.Split(strings.TrimSpace(lines[0]), " "), nil
 }
 
+func getUsedIPs(networkName string) ([]string, error) {
+	format := `{{range (index . "Containers")}}{{index . "IPv4Address"}} {{end}}`
+	cmd := exec.Command("docker", "network", "inspect", "-f", format, networkName)
+	lines, err := exec.OutputLines(cmd)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get used ip addresses")
+	}
+
+	usedIPs := strings.Split(strings.TrimSpace(lines[0]), " ")
+	for i := range usedIPs { // Trim the CIDR notation
+		usedIPs[i] = strings.Split(usedIPs[i], "/")[0]
+	}
+
+	return usedIPs, nil
+}
+
+func generateNodeLabels(node *config.Node) []string {
+	args := []string{}
+	for k, v := range node.ExtraLabels {
+		args = append(args, "--label", fmt.Sprintf("%s=%s", k, v))
```

Comment:
Any concerns around escaping either label or the value?

Reply:
Not particularly, docker will error out if it's an invalid label and be pretty obvious about it, best left to docker to handle it I think.